### PR TITLE
[ML] allow datafeeds to run if there are any concrete indices

### PR DESCRIPTION
**Pull request by benwtrent**
This PR allows a datafeed to be assigned to a node if only one index pattern has concrete indices.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedNodeSelector.java [hunk 1]
```diff
@@ -120,37 +121,34 @@ private AssignmentFailure checkAssignment() {
 
     @Nullable
     private AssignmentFailure verifyIndicesActive() {
-        for (String index : datafeedIndices) {
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedNodeSelector.java:123]: The other option is to do this foreach loop but simply continue if `allow_no_indices` is true. 

@dimitris-athanasiou @droberts195 what do you think? I am not sure if we want to do this with `allow_no_indices` being true as then we might have an issue later on when we attempt to extract fields. 

This way enforces at least one concrete index to exist (for now)

**Comment by benwtrent** [comment]
run elasticsearch-ci/packaging-sample-windows

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java [hunk 2]
```diff
@@ -146,6 +146,43 @@ public void testLookbackOnlyDataStream() throws Exception {
...
+        long twoWeeksAgo = oneWeekAgo - 604800000;
+        indexDocs(logger, "data-1", numDocs, twoWeeksAgo, oneWeekAgo);
+
+        Job.Builder job = createScheduledJob("lookback-empty-pattern-job");
+        registerJob(job);
+        putJob(job);
+        openJob(job.getId());
+        assertBusy(() -> assertEquals(getJobStats(job.getId()).get(0).getState(), JobState.OPENED));
+
+        List<String> t = new ArrayList<>(2);
```
Comment (davidkyle) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:165]: `t`? 
Whats wrong with `u` or `f` or `q`

I think a unit test would be easier to understand and provide better coverage.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedNodeSelector.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedNodeSelector.java:123]: I think it makes sense to require at least one concrete index for now.

**Review by davidkyle — APPROVED** [review]
LGTM

Thanks for the unit test

**Pull request merged**